### PR TITLE
Multiple broker services

### PR DESCRIPTION
**Pull request by DIOHz0r**
### Changes description

This PR is to allow the plugin to use multiple broker services as MQTT, Push, etc. The concept and use is based on the [symfony messenger component](https://symfony.com/doc/current/components/messenger.html) 

### Checklist

Please check if your PR fulfills the following specifications:

- [x] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@DIOHz0r |171|

### References

Closes #N/A
Related #776

### CODE REVIEW

File: inc/broker/brokerworker.class.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      }
+
+      $this->receiver->receive(function ($envelope) {
+         if (null === $envelope) {
+            return;
+         }
+         $this->bus->dispatch($envelope->with(new BrokerReceivedMessage()));
+      });
+   }
+}
```
[the remark this answers is not in the document]
Comment (DIOHz0r) [reply inc/broker/brokerworker.class.php]: Hi, the brokerworker class is used on the mqtt.php script. I tested it using the ```php -f mqtt.php``` to get the agent messages responses but no with the shell script for the flyvemdm service

[the remark this answers is not in the document]
Comment (btry) [reply inc/broker/brokerworker.class.php]: II tested with php mqtt.php and I SWA that the daemon quits immediately after being connected to the server.

The behaviour of xdebug seems strange to me and might be caused by a fork of the php process. This requires some more investigation

**Comment by ajsb85** [comment]
Hi,  @DIOHz0r 
Did you covered all the tests?
Do you need help for the code documentation?

**Comment by DIOHz0r** [comment]
There are some tests for the broker service and I tested the message sending process for agents and fleets, however, not all tests were programmed to do it on another PR because this one is really complicated to review.

**Pull request merged**